### DISCUSSION TITLE
does the multi-gpu trainer share the batches among the subprocesses

### DISCUSSION THREAD

**Discussion by b1nch3f** [coqui-ai/Trainer#53] [Q&A]
something is odd with the trainer. When I train on 2 GPUs on a dataset D, it takes X hours, if I train on 4 GPU it still takes X hours, I was expecting it to take X / 2 hours. Does the trainer share the batches between subprocesses or each subprocess takes it's own batch?

**Comment by erogol** [comment]
The same answer here https://github.com/coqui-ai/Trainer/discussions/54

Probably all GPUs see the all dataset in an epoch.
